python/pyspark/pandas/utils.py

```diff
@@ -927,14 +939,20 @@ def spark_column_equals(left: Column, right: Column) -> bool:
     >>> spark_column_equals(sdf1["x"] + 1, sdf2["x"] + 1)
     False
     """
-    return left._jc.equals(right._jc)
+    if isinstance(left, Column):
+        return left._jc.equals(right._jc)
+    elif isinstance(left, SparkConnectColumn):
+        return repr(left) == repr(right)
 
```

Comment:
should we raise an exception here if it's not a Column or SparkConnectColumn

Reply:
Yes, let me add a exception